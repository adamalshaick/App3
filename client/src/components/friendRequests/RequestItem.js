import React, { Component } from "react";
import PropTypes from "prop-types";
import { acceptRequest, declineRequest } from "../../actions/profileActions";
import { connect } from "react-redux";
import styled from "styled-components";
import { Link } from "react-router-dom";

const Request = styled.div`
  width: 300px;
  height: 150px;
  background-color: whitesmoke;
  border-radius: 1rem;
  position: absolute;
  top: 10px;
  right: 10px;
  z-index: 1;
  border: lightgrey solid 1px;
`;

export class RequestItem extends Component {
  onAccept(id) {
    this.props.acceptRequest(id);
  }

  onDecline(id) {
    this.props.declineRequest(id);
  }
  render() {
    const { request, profile, profiles } = this.props;
    const requestProfile = profiles.filter(
      profile => profile.user._id === request
    );
    return (
      <Request>
        <div className="text-center mt-4">
          <Link to={`/profile/${requestProfile[0].handle}`}>
            {requestProfile[0].handle}
          </Link>{" "}
          wants to be your friend
        </div>
        <div className="text-center mt-3">
          <button
            onClick={this.onAccept.bind(this, request._id)}
            className="btn btn-outline-success"
          >
            Accept
          </button>
          <button
            onClick={this.onDecline.bind(this, request._id)}
            className="btn btn-outline-danger ml-2"
          >
            Decline
          </button>
        </div>
      </Request>
    );
  }
}

RequestItem.propTypes = {
  request: PropTypes.object.isRequired,
  acceptRequest: PropTypes.func.isRequired,
  declineRequest: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { acceptRequest, declineRequest }
)(RequestItem);
