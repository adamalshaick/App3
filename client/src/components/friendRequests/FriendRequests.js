import React, { Component } from "react";
import RequestItem from "./RequestItem";
import PropTypes from "prop-types";

class FriendRequests extends Component {
  render() {
    const { profile, profiles } = this.props;
    return profile.friendRequests.map(request => (
      <RequestItem
        key={request._id}
        request={request}
        profile={profile}
        profiles={profiles}
      />
    ));
  }
}

FriendRequests.propTypes = {
  profile: PropTypes.object.isRequired
};

export default FriendRequests;
