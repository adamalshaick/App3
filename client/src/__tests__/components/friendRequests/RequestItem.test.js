import React from "react";
import { RequestItem } from "../../../components/friendRequests/RequestItem";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

let wrapper;

const Props = {
  request: {
    user: "user",
    handle: "handle",
    avatar: "avatar"
  },

  profile: {
    handle: "handle"
  },

  profiles: [
    {
      user: {
        _id: "user"
      },
      handle: "handle"
    },
    {
      user: {
        _id: "user2"
      },
      handle: "handle"
    }
  ]
};

wrapper = shallow(<RequestItem {...Props} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
