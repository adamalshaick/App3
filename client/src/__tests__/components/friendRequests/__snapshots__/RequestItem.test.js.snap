// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<styled.div>
  <div
    className="text-center mt-4"
  >
    <Link
      replace={false}
      to="/profile/handle"
    >
      handle
    </Link>
     
    wants to be your friend
  </div>
  <div
    className="text-center mt-3"
  >
    <button
      className="btn btn-outline-success"
      onClick={[Function]}
    >
      Accept
    </button>
    <button
      className="btn btn-outline-danger ml-2"
      onClick={[Function]}
    >
      Decline
    </button>
  </div>
</styled.div>
`;
